.github/workflows/ci.yaml

```
@@ -45,6 +45,39 @@ jobs:
           mkdir -p "$HOME/.config/claude/projects"
       - run: nix develop --command pnpm run test
 
+  code-coverage:
+    needs: changes
+    if: needs.changes.outputs.code-changed == 'true' && (github.event_name == 'pull_request' || github.ref == format('refs/heads/{0}', github.event.repository.default_branch))
+    runs-on: ubuntu-24.04-arm
+    permissions:
+      contents: read
+      code-quality: write
```

actionlint (1.7.12)

[error] unknown permission scope "code-quality". all available permission scopes are "actions", "artifact-metadata", "attestations", "checks", "contents", "deployments", "discussions", "id-token", "issues", "models", "packages", "pages", "pull-requests", "repository-projects", "security-events", "statuses"

(permissions)